Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -474,6 +479,8 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
                 _analyzer.AddModuleAlias(first, a);
             }
 
+            AddParentModules(first, path);
```

Comment:
Yes, I can add comments. The issue and project structure is here https://github.com/Microsoft/PTVS/issues/4138. Folders up the tree don't have `__init__.py` and we miss them.

Reply:
Namespace packages should be resolved downwards from our search paths. We can't just resolve them upwards or we'll never know where to stop generating package names.

We can assume the root of the workspace is a search path (and we should already have that information available). Otherwise, the IDE should provide any other search paths explicitly.